modules/hooks/activeEffects.js

```diff
@@ -1,22 +1,22 @@
 
 export default function () {
 
-    Hooks.on("createActiveEffect", (effect, options, id) => {_runUpdateEffects(effect, "create", options, id)})
-    Hooks.on("updateActiveEffect", (effect, options, id) => {_runUpdateEffects(effect, "update", options, id)})
-    Hooks.on("deleteActiveEffect", (effect, options, id) => {
+    Hooks.on("createActiveEffect", async (effect, options, id) => {await _runUpdateEffects(effect, "create", options, id)})
+    Hooks.on("updateActiveEffect", async (effect, options, id) => {await _runUpdateEffects(effect, "update", options, id)})
+    Hooks.on("deleteActiveEffect", async (effect, options, id) => {
         if (effect.parent.documentName == "Actor")
         {
             let items = effect.parent.items.filter(i => i.getFlag("wfrp4e", "fromEffect") == effect.id);
             if (items.length)
             {
                 ui.notifications.notify(game.i18n.format("EFFECT.DeletingItems", {items : items.map(i => i.name).join(", ")}))
-                effect.parent.deleteEmbeddedDocuments("Item", items.map(i => i.id));
+                await effect.parent.deleteEmbeddedDocuments("Item", items.map(i => i.id));
             }
         }
-        _runUpdateEffects(effect, "delete", options, id)
+        await _runUpdateEffects(effect, "delete", options, id)
     })
 
-    Hooks.on("preCreateActiveEffect", (effect, data, options, id) => {
+    Hooks.on("preCreateActiveEffect", async (effect, data, options, id) => {
```

Comment:
Hooks should not be async

Reply:
i think it depends on the hook (if it can stop the method execution chain, maybe the pre hooks shouldn't be? I haven't look to deep, but here are examples of async hooks: https://foundryvtt.wiki/en/development/guides/Hooks_Listening_Calling

Reply:
Right, many hooks could be async, but `preCreate` hooks shouldn't be for the reason you gave, and more importantly data modifications might get lost (not applied in time). While it may currently work fine it's better to keep it sync so it's remembered and enforced for future code changes.